actionpack/lib/action_dispatch/http/cache.rb

```diff
@@ -133,7 +133,7 @@ def generate_weak_etag(validators)
         end
 
         def generate_strong_etag(validators)
-          %("#{Digest::MD5.hexdigest(ActiveSupport::Cache.expand_cache_key(validators))}")
+          %("#{ActiveSupport::Digest.hexdigest(ActiveSupport::Cache.expand_cache_key(validators))}")
```

Comment:
I'm not sure it's a good idea for the length of the ETag header to reveal which digest algorithm is configured.

When I suggested truncating the output of `hexdigest`, I meant inside our wrapper function; we shouldn't burden callers with deciding whether they need a fixed length string.

Of course, switching to SHA-1 everywhere as @matthewd suggested would also resolve this issue.